### PR TITLE
Force-close channels on reorg only if the funding is unconfirmed

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Currently, if a channel's funding is locked in and then later
reorg'd back to half of the channel's minimum-depth we will
immediately force-close the channel. However, this can happen at
the fork-point while processing a reorg, and generally reorgs do
not reduce the block height at all, making this a rather useless
endeavor.

Ideally we'd never auto-force-close channels at all due to a reorg,
instead simply marking it as inactive until the funding
transaction is re-confirmed (or allowing the user to attempt to
force-close or force-closing once we're confident we have
completed reorg processing if we're at risk of losing funds
already received in the channel).

Sadly, we currently do not support changing a channel's SCID and
updating our SCID maps, so we cannot yet remove the automated
force-close logic. Still, there is no reason to do it until a
funding transaction has been removed from the chain.

This implements that change - only force-closeing once a channel's
funding transaction has been reorg'd out (still potentially at a
reorg's fork point). This continues to imply a 1-confirmation
channel will always be force-closed after a reorg of the funding
transaction, and will imply a similar behavior with 0-conf
channels.

### CODE REVIEW

**Review by vincenzopalazzo — COMMENTED** [review]
LGTM, maybe can be a good introduce atest to cover this edge case?

File: lightning/src/ln/reorg_tests.rs [hunk 1]
```diff
@@ -212,11 +212,7 @@ fn do_test_unconf_chan(reload_node: bool, reorg_after_reload: bool, use_funding_
 		} else {
 			disconnect_all_blocks(&nodes[0]);
 		}
-		if connect_style == ConnectStyle::FullBlockViaListen && !use_funding_unconfirmed {
```
Comment (valentinewallace) [inline lightning/src/ln/reorg_tests.rs:215]: I don't think the `connect_style` parameter is used anymore in this test, everything passes when it's `unused`

Comment (valentinewallace) [reply lightning/src/ln/reorg_tests.rs:215]: All tests pass with that line commented out

**Review by valentinewallace — APPROVED** [review]

**Review by vincenzopalazzo — APPROVED** [review]
LGTM

**Review by valentinewallace — DISMISSED** [review]

**Review by dunxen — APPROVED** [review]
cr-ACK  1568097

**Review by arik-so — APPROVED** [review]

**Review by valentinewallace — APPROVED** [review]

**Pull request merged**